gson/src/test/java/com/google/gson/functional/MapTest.java

```diff
@@ -210,6 +213,49 @@ public void testMapDeserializationWithUnquotedLongKeys() {
     assertThat(map.get(longKey)).isEqualTo("456");
   }
 
+  @Test
+  public void testMapStringKeyDeserialization() {
```

Comment:
There is some overlap between the newly added tests here and in `ConstructorConstructorTest`, should we keep both nonetheless?